functions.php

```diff
@@ -320,130 +322,141 @@ function wp_autoupdates_plugins_bulk_actions( $actions ) {
+			wp_die( __( 'Please connect to your network admin to manage plugins automatic updates.', 'wp-autoupdates' ) );
+		}
 
-	if ( is_multisite() && ! is_network_admin() ) {
-		wp_die( __( 'Please connect to your network admin to manage plugins automatic updates.', 'wp-autoupdates' ) );
-	}
+		check_admin_referer( 'updates' );
+
+		if ( empty( $plugin ) ) {
+			wp_redirect( self_admin_url( "plugins.php?plugin_status=$status&paged=$page&s=$s" ) );
+			exit;
+		}
+
+		$auto_updates = get_site_option( 'wp_auto_update_plugins', array() );
+
+		$auto_updates[] = $plugin;
+		$auto_updates   = array_unique( $auto_updates );
 
-	$plugin = ! empty( esc_html( $_GET['plugin'] ) ) ? wp_unslash( esc_html( $_GET['plugin'] ) ) : '';
-	$page   = ! empty( $_GET['paged'] ) ? absint( $_GET['paged'] ) : '';
-	$status = isset( $_GET['plugin_status'] ) && ! empty( esc_html( $_GET['plugin_status'] ) ) ? wp_unslash( esc_html( $_GET['plugin_status'] ) ) : '';
-	$s      = isset( $_GET['s'] ) && ! empty( esc_html( $_GET['s'] ) ) ? wp_unslash( esc_html( $_GET['s'] ) ) : '';
+		update_site_option( 'wp_auto_update_plugins', $auto_updates );
 
-	if ( empty( $plugin ) ) {
-		wp_redirect( self_admin_url( "plugins.php?plugin_status=$status&paged=$page&s=$s" ) );
+		wp_redirect( self_admin_url( "plugins.php?disable-auto-update=true&plugin_status=$status&paged=$page&s=$s" ) );
 		exit;
-	}
+	} elseif ( 'disable-auto-update' === $_GET['action'] ) {
+		if ( ! current_user_can( 'update_plugins' ) || ! wp_autoupdates_is_plugins_auto_update_enabled() ) {
+			wp_die( __( 'Sorry, you are not allowed to disable plugins automatic updates.', 'wp-autoupdates' ) );
+		}
 
-	$wp_auto_update_plugins = get_site_option( 'wp_auto_update_plugins', array() );
+		if ( is_multisite() && ! is_network_admin() ) {
+			wp_die( __( 'Please connect to your network admin to manage plugins automatic updates.', 'wp-autoupdates' ) );
+		}
 
-	if ( 'disable-auto-update' === $_GET['action'] ) {
-		$wp_auto_update_plugins = array_diff( $wp_auto_update_plugins, array( $plugin ) );
-		$action_type            = 'disable-auto-update=true';
-	} else {
-		$wp_auto_update_plugins[] = $plugin;
-		$wp_auto_update_plugins   = array_unique( $wp_auto_update_plugins );
-		$action_type = 'enable-auto-update=true';
-	}
+		check_admin_referer( 'updates' );
+
+		if ( empty( $plugin ) ) {
+			wp_redirect( self_admin_url( "plugins.php?plugin_status=$status&paged=$page&s=$s" ) );
+			exit;
+		}
 
-	update_site_option( 'wp_auto_update_plugins', $wp_auto_update_plugins );
-	wp_redirect( self_admin_url( "plugins.php?$action_type&plugin_status=$status&paged=$page&s=$s" ) );
-	exit;
+		$auto_updates = get_site_option( 'wp_auto_update_plugins', array() );
+
+		$auto_updates = array_diff( $auto_updates, array( $plugin ) );
+
+		update_site_option( 'wp_auto_update_plugins', $auto_updates );
+
+		wp_redirect( self_admin_url( "plugins.php?disable-auto-update=true&plugin_status=$status&paged=$page&s=$s" ) );
```

Comment:
Just a nitpick but when concatenating "inline", it is usually more readable to wrap the variables in `{}`. Then the above would be:
`"plugins.php?disable-auto-update=true&plugin_status={$status}&paged={$page}&s={$s}"`

At the same time, this can be done "the WP way" by passing an array to `add_query_arg()` too :)

Reply:
I agree 100%.

However, the string is exactly how all of the equivalent ones in `wp-admin/plugins.php` are constructed, e.g, https://core.trac.wordpress.org/browser/trunk/src/wp-admin/plugins.php#L38.

When creating the core merge patch I've been trying to do equivalent things in *exactly* the same way as existing core code (which is why the way the redirect URL is constructed in `wp_autoupdates_handle_plugins_enable_disable()` is different than in wp_autoupdates_handle_themes_enable_disable()`).

Reply:
> I've been trying to do equivalent things in exactly the same way as existing core code

Yeah, I understand. But not all of the core code is necessarily written in the best way possible. There's always some room for improvement. Perhaps after merging this, we can open a ticket and fix the old core code too? Looking there, most of these were added over 11 years ago (https://core.trac.wordpress.org/changeset/11029), that's in the PGP 4.x era :)

Reply:
Agreed, opening a ticket after merge was going to be my suggestion as well.

Remember that conversation we were having in Slack about inconsistencies between list tables?  It also applies to the "driver" files for the screens on which list tables appear :-(